File: 1295 Find Numbers with Even Number of Digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int cnt = 0;
        for (int num : nums) {
            if (String.valueOf(num).length() % 2 == 0) {
                ++cnt;
            }
        }
        return cnt;
    }
}

/* solution using dividing by 10
class Solution {
    public int findNumbers(int[] nums) {
        int cnt = 0;
        for (int num : nums) {
            boolean isEven = true;
            while (num > 0) {
                num /= 10;
                isEven = !isEven;
            }
            if (isEven) {
                ++cnt;
            }
        }
        return cnt;
    }
}
*/

File: 1295 Find Numbers with Even Number of Digits.py
class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        cnt = 0
        for num in nums:
            if len(str(num)) % 2 == 0:
                cnt += 1
        return cnt
''' solution using dividing by 10
class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        cnt = 0
        for num in nums:
            isEven = True
            while num > 0:
                num //= 10
                isEven = not isEven
            if isEven:
                cnt += 1
        return cnt
'''